peer/interfaces.go

```diff
@@ -64,3 +66,25 @@ type MessageLink interface {
 	// HandleChannelUpdate passes lnwire.Message to the MessageLink.
 	HandleChannelUpdate(lnwire.Message)
 }
+
+// MessageSwitch is an interface that manages setup, retrieval, and shutdown of
+// MessageLink implementations.
+type MessageSwitch interface {
```

Comment:
The motivation is to make `peer` not rely on `htlcswitch.Switch`, so instead use a `ChannelSwitch` that can be replaced with a mock.  Otherwise you need to provide a `*htlcswitch.Switch` when testing

Reply:
channelSwitch needs to wrap `GetLink` to return `MessageLink` as `*htlcswitch.Switch` returns `ChannelLink`

Reply:
If we want to be able to replace `*htlcswitch.Switch` in peer tests (which IMO is worth it, as it does not belong in the peer unit tests):
- Need an interface to pass into the peer config instead of the raw Switch pointer.
- The interface should also live in the peer package.
    - So, the Switch now has to comply with this peer-pkg interface.
- Peer calls the interface's `GetLink` which returns `ChannelLink, error`
    - To implement this interface in unit tests, and have a non-nil `ChannelLink`** returned on `GetLink`, we must implement this interface in the peer.
    - `ChannelLink` interface has this method `HandleSwitchPacket(*htlcPacket) error` which uses a private `*htlcPacket` argument.
- So there are two options:
    - Change the interface `GetLink` to not return `ChannelLink, error`
    - Change `ChannelLink` methods to not have `*htlcPacket`

So IMO this is a workaround until we can cleanup the htlcswitch package.  One thing would be to remove `*htlcPacket` in public methods.  And another would be to have `GetLink` return a concrete type so that it can fit into any interface because we don't have interfaces narrowing `ChannelLink` -> `MessageLink`.

** I have a test-case that requires a non-nil link to be returned.

Reply:
Is this a blocker for your ACK? I can figure out the best step to avoid this "type confusion"

Reply:
The tests do not use `NewChannelSwitch`, they use the `mockMessageSwitch`?  The peer interface will always be a subset of functionality provided by the `Switch` and only cares about the functions used in the `peer` package so I think this is achieved?  I think why I didn't change `*htlcPacket` was because it was a pretty large change that touched the unit tests of the `htlcswitch` package